optuna/importance/_mutual_information/_evaluator.py

```diff
@@ -0,0 +1,120 @@
+            params = list(search_space.keys())
+
+        values, scores = _get_multivariate_observation_pairs(study, params)
+        parameters = {k: np.array(v) for k, v in values.items()}
+        scores = np.array([score for step, score in scores])
+
+        score_range = np.max(scores) - np.min(scores)
+        score_distribution = UniformDistribution(
+            np.min(scores) - 0.3 * score_range, np.max(scores) + 0.3 * score_range
+        )  # type: BaseDistribution
+        score_space = {"score": score_distribution}
+
+        if self._n_MC is None:
+            mpe_score = _MultivariateParzenEstimator({"score": scores}, score_space)
+            score_log_prob = mpe_score.log_pdf({"score": scores})
+
+            mi = {}
+            for param_name, param_values in parameters.items():
+                # We construct parameter space.
+                param_space = {param_name: search_space[param_name]}
+                joint_space = deepcopy(param_space)
+                joint_space.update(score_space)
+                # We construct Parzen estimators.
+                mpe_param = _MultivariateParzenEstimator({param_name: param_values}, param_space)
+                mpe_joint = _MultivariateParzenEstimator(
+                    {param_name: param_values, "score": scores}, joint_space
+                )
+                # We calculate log_probs.
+                param_log_prob = mpe_param.log_pdf({param_name: param_values})
+                joint_log_prob = mpe_joint.log_pdf({param_name: param_values, "score": scores})
+
+                mi[param_name] = np.mean(joint_log_prob - score_log_prob - param_log_prob)
+        else:
+            mi = {}
+            for param_name, param_values in parameters.items():
+                # We construct parameter space.
+                param_space = {param_name: search_space[param_name]}
+                joint_space = deepcopy(param_space)
+                joint_space.update(score_space)
+                # We construct Parzen estimators.
+                mpe_score = _MultivariateParzenEstimator({"score": scores}, score_space)
+                mpe_param = _MultivariateParzenEstimator({param_name: param_values}, param_space)
+                mpe_joint = _MultivariateParzenEstimator(
+                    {param_name: param_values, "score": scores}, joint_space
+                )
+                # We get Monte Carlo samples.
+                rng = np.random.RandomState()
+                mc_samples = mpe_joint.sample(rng, self._n_MC)
+                # We calculate log_probs.
+                score_log_prob = mpe_score.log_pdf({"score": mc_samples["score"]})
+                param_log_prob = mpe_param.log_pdf({param_name: mc_samples[param_name]})
+                joint_log_prob = mpe_joint.log_pdf(mc_samples)
+
+                mi[param_name] = np.mean(joint_log_prob - score_log_prob - param_log_prob)
+
+        mi = {k: v for k, v in sorted(mi.items(), key=lambda item: item[1], reverse=True)}
```

Comment:
If the order of `mi` is important, we should use `OrderedDict` as dicts of some old-ish Python do not keep insertion order.

Reply:
I do not think the order of `mi` matters here, but I appreciate your caution. I might make such a mistake somewhere else without it.

Reply:
Ah, yes, I now understood your point. The order actually matters in the last line. I will fix it. Thank you!